### PR TITLE
docs: add a migration guide for Kubeflow Trainer users

### PR DESCRIPTION
**Pull request by apurvapatkeshwar**
**What type of PR is this? (check all applicable)**
- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [x] Documentation Update

**What changed?**

Adds `docs/user-guides/migration/migrate-from-kubeflow-trainer.md`, the second guide in the migration section started by #1783 (both were asked for in #1705). It is aimed at teams running `PyTorchJob`/`TFJob`/`TrainJob` who want to know exactly what changes: training-job CRDs are replaced by Uniflow tasks with a `RayTask` resource declaration, and distribution comes from the Ray Train-based trainers (`LightningTrainer`, `XGBoostTrainer`). Follows the same skeleton the KubeRay guide established: concept-mapping table, a real before/after (PyTorchJob YAML to a pipeline task), what you gain, and an explicit "what does not map yet" section.

**Why?**

#1705 asked for migration guides from Kubeflow Trainer and KubeRay. #1783 covers KubeRay; this covers Kubeflow Trainer. Teams evaluating a move currently have to reverse-engineer the trainer library and the Ray plugin to learn that (for example) there is no TensorFlow trainer and no gang scheduling -- this states it plainly.

**How did you test it?**

Docs-only change; testing here means accuracy and link integrity:

- Every capability claim was verified against the code before writing, and re-verified against current main before posting: the example uses the actual `LightningTrainerParam` field names (`create_model_fn`, `create_model_fn_kwargs`, `train_data`, `val_data`, `batch_size`) rather than the incorrect names some existing docs carry (see #1787); the gaps section (no training-job CRD equivalent, no TensorFlow trainer, fixed-size clusters on both paths, no gang scheduling, Ray pinned to 2.3.1 on the task path) matches the plugin and jobs-client behavior at HEAD.
- All relative links and heading anchors resolve against this branch's docs tree (the KubeRay guide link resolves because of the stack).
- Filename, frontmatter, admonitions, and link style follow `docs/contributing/documentation-guide.md`.
- The docs build was not run locally (no bun on this machine) -- flagging so reviewers know to lean on the docs CI job.

**Potential risks**

None at runtime -- documentation only. The main risk is factual drift if the trainer library or Ray plugin changes; the gaps section states current behavior plainly rather than implying anything is scheduled, so it fails safe.

**Breaking Changes**

- [x] No breaking changes
- [ ] API changes (Go exported symbols or function signatures, Python public functions or classes)
- [ ] Proto changes (enum value renumbering, field number changes, field removal, service removal)
- [ ] Helm changes (new required values, renamed or removed keys, changed value semantics)
- [ ] Config/deployment changes (new required env vars, renamed container args, changed ports or mount paths)

**Release notes**

Docs only; nothing operational.

**Documentation Changes**

This PR is the documentation change: one new page under `docs/user-guides/migration/`.

### CODE REVIEW

File: docs/user-guides/migration/migrate-from-kubeflow-trainer.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+There is no `Master`/`Worker` role split anywhere in that. Ray Train coordinates workers internally, and rank 0 is a runtime concept rather than a differently-shaped Kubernetes object.
+
+## Concept mapping
+
+| Concept | Kubeflow Trainer | Michelangelo AI |
+|---|---|---|
+| **Training job resource** | `PyTorchJob` / `TFJob` / `TrainJob` (CRDs) | None — a Uniflow task with a `RayTask` config |
+| **Distributed PyTorch** | `PyTorchJob` replica specs | `LightningTrainer`, a Ray Train `TorchTrainer` subclass |
+| **Distributed XGBoost** | `XGBoostJob` | `XGBoostTrainer`, a Ray Train `XGBoostTrainer` subclass |
```
Comment (andriika) [inline docs/user-guides/migration/migrate-from-kubeflow-trainer.md:32]: I believe Michelangelo doesn't currently provide an `XGBoostTrainer` (a subclass of Ray Train's `XGBoostTrainer`), unless I'm missing something.

Could you please double-check that?

Comment (apurvapatkeshwar) [reply docs/user-guides/migration/migrate-from-kubeflow-trainer.md:32]: Double checked - it does exist;  it only landed last week in #1793 so easy to miss 
`XGBoostTrainer` in `michelangelo.lib.trainer.xgboost` (subclass of `ray.train.xgboost.XGBoostTrainer`)
Ref: https://github.com/michelangelo-ai/michelangelo/blob/main/python/michelangelo/lib/trainer/xgboost/xgboost_trainer.py#L157
